packages/react/src/core/utils/useFocus.ts

```diff
@@ -0,0 +1,40 @@
+import { ForwardedRef, useEffect, useRef } from 'react';
+
+export type UseFocusParams<T> = {
+	disabled?: boolean;
+	focusOnUpdate?: ReadonlyArray<unknown>;
+	forwardedRef?: ForwardedRef<T>;
+};
+
+/**
+ * useFocus
+ *
+ * Optionally takes a React.Ref (or creates a fallback) to be assigned
+ * to an element you wish to be focused when an array of values are
+ * updated or once when it is mounted.
+ *
+ * @param disabled - Optionally disable the focusing.
+ * @param focusOnUpdate - Optionally define an array of values, that when changed, will trigger focus. Note: by leaving this undefined focus will be called once on mount.
+ * @param forwardedRef - Optionally use a forwarded ref. If a ref isn't provided, a fallback ref will be created.
+ * @returns React.Ref to be assigned to the element to be focused.
+ */
+
+export function useFocus<T extends HTMLElement>({
+	disabled,
+	focusOnUpdate = [],
+	forwardedRef,
+}: UseFocusParams<T> = {}) {
+	const fallbackRef = useRef<T>(null);
+	const ref = forwardedRef || fallbackRef;
+
+	useEffect(() => {
+		if (disabled || typeof ref === 'string' || !('current' in ref)) {
+			return;
+		}
+
+		ref?.current?.focus?.();
+		// We are allowing a non-literal array of dependencies to enable consumers to trigger focus updates based on a list of their own trigger-values.
+	}, [disabled, focusOnUpdate]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
So changing the `disabled` state also immediately focuses? When do you we use `disabled`?

Reply:
>So changing the disabled state also immediately focuses?

Hmm good question. It does but I think that's ok.

>When do you we use disabled?

You can't conditionally call hooks, so the `disabled` prop is primarily there to prevent focus in the two alerts when it's not turned on by consumers via `focusOnMount` and `focusOnUpdate`.